registry/remote/registry_test.go

```diff
@@ -193,3 +193,42 @@ func TestRegistry_Repository(t *testing.T) {
 		t.Errorf("Registry.Repository() = %v, want %v", got, want)
 	}
 }
+
+// This is a package-level example:
+func ExampleRepositories() {
+	var sampleRepoSet = []string{"public/repo1", "public/repo2", "internal/repo3"}
+	// Mocking local registry
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		result := struct {
+			Repositories []string `json:"repositories"`
+		}{
+			Repositories: sampleRepoSet,
+		}
+		json.NewEncoder(w).Encode(result)
+	}))
+	defer ts.Close()
+	uri, err := url.Parse(ts.URL)
+	if err != nil {
+		panic(err)
+	}
+
+	// Example: List repositories in a registry
```

Comment:
Does the go doc start at line 215 or should the setup be moved into testmain. Asking because I don’t know enough about godocs.

Reply:
The go doc shows the whole function, with the mocking part. I have moved the mocking function out. Now the doc looks like